centrallix/expression/exp_compiler.c

```diff
@@ -1072,16 +1072,10 @@ expBindExpression(pExpression exp, pParamObjects objlist, int flags)
 			break;
 			}
 		    }
-		if (exp->ObjID == -1)
-		    {
-		    cm |= EXPR_MASK_EXTREF;
-		    }
-		}
-	    else if (exp->ObjID == -2 || exp->ObjID == -3)
-		{
-		if (exp->ObjID == -2) cm |= (1<<(objlist->CurrentID));
-		if (exp->ObjID == -3) cm |= (1<<(objlist->ParentID));
+		cm |= EXPR_MASK_EXTREF;
 		}
+	    else if (exp->ObjID == EXPR_CTL_CURRENT) cm |= (1<<(objlist->CurrentID));
+	    else if (exp->ObjID == EXPR_CTL_PARENT) cm |= (1<<(objlist->ParentID));
```

Comment:
Good change to remove the magic numbers, but use EXPR_OBJID_CURRENT and EXPR_OBJID_PARENT, respectively.